metricbeat/module/pulsar/broker/broker.go

```diff
@@ -0,0 +1,94 @@
+package broker
+
+import (
+	p "github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+//Broker MetricSet
+type MetricSet struct{
+	mb.BaseMetricSet
+	prometheus p.Prometheus
+	mapping *p.MetricsMapping
+}
+
+//New returns a prometheus based metricset
+func New(base mb.BaseMetricSet) (mb.MetricSet, error){
+        prometheus, _ := p.NewPrometheusClient(base)
+
+	return &MetricSet{
+                BaseMetricSet: base,
+                prometheus:    prometheus,
+                mapping: &p.MetricsMapping{
```

Comment:
There seem to be some labels in the prometheus response that seem to be be valuable as metadata here.

For example from this metric Metricbeat could obtain the cluster name, the namespace and the topic name.
```
 pulsar_consumers_count{cluster="standalone",namespace="public/functions",topic="persistent://public/functions/assignments"} 0.0 1621630704621
```

This could be done by adding labels to the mapping, something like this:
```
                Labels: map[string]p.LabelMap{
                        "cluster":   p.KeyLabel("cluster.name"),
                        "namespace": p.KeyLabel("namespace.name"),
                        "topic":     p.KeyLabel("topic.name"),                     
                },
```

You can see examples of this in the state metricsets of the kubernetes module, like here: https://github.com/elastic/beats/blob/df616d76b04c4355581cd6a22141458d49d0d6f1/metricbeat/module/kubernetes/state_pod/state_pod.go#L50-L57